server/server.go

```diff
@@ -407,6 +407,7 @@ func createQueryResolver(
 		blockingResolver,
 		resolver.NewCachingResolver(cfg.Caching, redisClient),
 		resolver.NewRewriterResolver(cfg.Conditional.RewriteConfig, conditionalUpstreamResolver),
+		resolver.NewSudnResolver(),
```

Comment:
We can move this resolver to the begin of the chain (like fqdnOnly). There is no need to cache response from this resolver or to apply blocking rules

Reply:
We can't move it before custom an conditional as these can possibly return valid responses for such requests.
This resolver is intended to stop those requests from reaching unconditional upstream resolvers.

Moving it to the start would break conditional ARPA responses.